src/js/tech/html5.js

```diff
@@ -115,6 +120,61 @@ class Html5 extends Tech {
   }
 
   /**
+   * When a captions track is enabled in the iOS Safari native player, all other
+   * tracks are disabled (including metadata tracks), which nulls all of their
+   * associated cue points. This will restore metadata tracks to their pre-fullscreen
+   * state in those cases so that cue points are not needlessly lost.
+   *
+   * @private
+   */
+  saveAndRestoreMetadataTrackModeInIOSNativePlayer_() {
+    const textTracks = this.textTracks();
+    let metadataTracksPreFullscreenState;
+
+    // captures a snapshot of every metadata track's current state
+    const takeMetadataTrackSnapshot = () => {
+      metadataTracksPreFullscreenState = [];
+
+      for (let i = 0; i < textTracks.length; i++) {
+        const track = textTracks[i];
+
+        if (track.kind === 'metadata') {
+          metadataTracksPreFullscreenState.push({
+            track,
+            storedMode: track.mode
+          });
+        }
+      }
+    };
+
+    // snapshot each metadata track's initial state, and update the snapshot
+    // each time there is a track 'change' event
+    takeMetadataTrackSnapshot();
+    textTracks.on('change', takeMetadataTrackSnapshot);
+
+    // when we enter fullscreen playback, restore all track modes to their initial state
+    this.on('fullscreenchange', (event, data) => {
```

Comment:
Optionally, if you listen to the `webkitenterfullscreen` and `webkitexitfullscreen` events, you wouldn't need to do an extra `isFullscreen` check but also you'd be able to remove the user agent check above.